src/main/java/com/feedzai/commons/sql/abstraction/ddl/DbFk.java

```diff
@@ -258,11 +257,17 @@ public Builder addReferencedColumns(final Collection<String> columns) {
 
         @Override
         public DbFk build() {
-            Preconditions.checkNotNull(referencedTable, "The referenced table can't be null.");
-            Preconditions.checkArgument(localColumns.size() == referencedColumns.size(),
-                    "The number of local columns and referenced columns must be the same. local: %s ; referenced: %s",
-                    localColumns, referencedColumns
-            );
+            Objects.requireNonNull(this.referencedTable, "The referenced table can't be null");
+
+            if (this.localColumns != this.referencedColumns) {
+                throw new IllegalArgumentException (
+                    "The number of local columns and referenced columns must be the same. local: "
+                        + this.localColumns
+                        + " ; referenced: "
+                        + this.referencedColumns
```

Comment:
This code format look weird. I'd rather use `String.format`, but it is up to you.